### ISSUE TITLE
the training is stuck at.... any idea on this issue

### ISSUE THREAD

**Issue by loveJasmine** [ufal/neuralmonkey#472]
^@2017-06-19 10:47:40: The model has 27 trainable variables:

                                 Variable name                                         Shape           Size

sentence_encoder/input_projection/word_embeddings:0                             [8004, 300]            2401200
sentence_encoder/bidirectional_rnn/fw/OrthoGRUCell/gates/weights:0              [600, 600]              360000
sentence_encoder/bidirectional_rnn/fw/OrthoGRUCell/gates/biases:0               [600]                      600
sentence_encoder/bidirectional_rnn/fw/OrthoGRUCell/candidate/weights:0          [600, 300]              180000
sentence_encoder/bidirectional_rnn/fw/OrthoGRUCell/candidate/biases:0           [300]                      300
sentence_encoder/bidirectional_rnn/bw/OrthoGRUCell/gates/weights:0              [600, 600]              360000
sentence_encoder/bidirectional_rnn/bw/OrthoGRUCell/gates/biases:0               [600]                      600
sentence_encoder/bidirectional_rnn/bw/OrthoGRUCell/candidate/weights:0          [600, 300]              180000
sentence_encoder/bidirectional_rnn/bw/OrthoGRUCell/candidate/biases:0           [300]                      300
decoder/initial_state/encoders_projection/weights:0                             [600, 256]              153600
decoder/initial_state/encoders_projection/biases:0                              [256]                      256
decoder/word_embeddings:0                                                       [8004, 300]            2401200
decoder/state_to_word_W:0                                                       [256, 8004]            2049024
decoder/state_to_word_b:0                                                       [8004]                    8004
decoder/attention_sentence_encoder/AttnW:0                                      [1, 1, 600, 600]        360000
decoder/attention_sentence_encoder/AttnV:0                                      [600]                      600
decoder/attention_sentence_encoder/AttnV_b:0                                    []                           1
decoder/attention_decoder/LinearProjection/LinearProjection/weights:0           [900, 300]              270000
decoder/attention_decoder/LinearProjection/LinearProjection/biases:0            [300]                      300
decoder/attention_decoder/OrthoGRUCell/gates/weights:0                          [556, 512]              284672
decoder/attention_decoder/OrthoGRUCell/gates/biases:0                           [512]                      512
decoder/attention_decoder/OrthoGRUCell/candidate/weights:0                      [556, 256]              142336
decoder/attention_decoder/OrthoGRUCell/candidate/biases:0                       [256]                      256
decoder/attention_decoder/attention_sentence_encoder/Attention/weights:0        [256, 600]              153600
decoder/attention_decoder/attention_sentence_encoder/Attention/biases:0         [600]                      600
decoder/attention_decoder/AttnOutputProjection/AttnOutputProjection/weights:0   [856, 256]              219136
decoder/attention_decoder/AttnOutputProjection/AttnOutputProjection/biases:0    [256]                      256

2017-06-19 10:47:40: Total number of all parameters: 9527353
2017-06-19 10:47:43: Initializing TensorBoard summary writer.
2017-06-19 10:47:47: TensorBoard writer initialized.
2017-06-19 10:47:47: Starting training

2017-06-19 10:47:47: Epoch 1 starts

.........

2017-06-19 11:01:14: Epoch 1/11  Instances 197010  series_named_greedy/runtime_xent: 0.06825    series_named_greedy/BLEU-4: 0.7618    series_named_greedy/train_xent: 0.04826    series_named_greedy/TER: 1.133

the GPU is running, a lot of GPU memory & CPU memory also available..

**Comment by jlibovicky** [comment]
My guess is the training reached the first validation and your validation data are just too big.

**Comment by tomkocmi** [comment]
Another issue could be if you are using "report_gpu_memory_consumption=True" in the tf_manager. This option in previous versions of NM was making similar problems.

**Comment by loveJasmine** [comment]
Actually, I got a 10 millions Question Answer Pairs dataset,
so, I followed the principal that split the the dataset as train/validate/test as ratio to 7/2/1
so, approximately 7 million line QA pairs as training set , 200 millions QA pairs as validation set

**Comment by tomkocmi** [comment]
The standard ratio split 7/2/1 is obsolete (there was a nice article about that by Andrew Ng, but I cannot find it). In your split, that mean your precision over the validation set will be roughly with a precision of 0.000001%. That's probably not that necessary precision. When we are doing machine translation we use a validation set of 3000 sentence pairs.

**Comment by jlibovicky** [comment]
The estimated time depends on the hardware you use. You can estimate the time for one batch: take the time between the training logs and divided it by `logging_period` from the configuration file. In this way, you can estimate the time you are going to need for 2M (I guess 200 was a typo) validation examples. It will be probably hours.

**Comment by loveJasmine** [comment]
yes, you are right @jlibovicky 
It keep moving after several hours

**Comment by loveJasmine** [comment]
@tomkocmi  I will try your proposal, thanks!
